src/app/home/home.component.html

```diff
@@ -36,41 +36,27 @@
   </a>
   <a routerLink="/notifications" mat-menu-item>See All</a>
 </mat-menu>
-
+<!--Left sidebar-->
 <mat-sidenav-container class="main-content" #content>
   <mat-sidenav #sidenav mode="side" class="primary-color main-sidenav nav-shadow" opened="true" [@sidenavState]="sidenavState" (@sidenavState.done)="endAnimation()">
+    <!--Top part, user content etc.-->
+    <!-- Less readable prevous class setting method: -->
+    <!-- [ngClass]="{'active': rlaRef.isActive, 'pulsate': route.title === pulsatingClassed && !rla.isActive}" -->
     <ul class="top-nav">
-      <li>
-        <a mat-button routerLink="/" i18n-title title="Home">
-          <mat-icon svgIcon="home"></mat-icon>
-          <label i18n>Home</label>
-        </a>
-      </li>
-      <li>
-        <a mat-button routerLink="/resources" i18n-title title="Resources">
-         <mat-icon svgIcon="myLibrary"></mat-icon>
-         <label i18n>Library</label>
-        </a>
-      </li>
-      <li>
-        <a mat-button routerLink="/courses" i18n-title title="Courses">
-          <mat-icon svgIcon="myCourses"></mat-icon>
-          <label i18n>Courses</label>
-        </a>
-      </li>
-      <li>
-        <a mat-button routerLink="/meetups" i18n-title title="Meetups">
-          <mat-icon svgIcon="myMeetups"></mat-icon>
-          <label i18n>Meetups</label>
-        </a>
-      </li>
-      <li>
-        <a mat-button routerLink="/users" i18n-title title="Users">
-          <mat-icon svgIcon="myTeams"></mat-icon>
-          <label i18n>Members</label>
+      <li *ngFor="let route of topNavEls">

```

Comment:
Each navigation link was written out individually because the `i18n` translation does not currently work to translate JavaScript in the component.  This will hopefully be updated for v6, but until then we should keep the navigation in a translatable state.

You've added a lot of attributes here, so I wouldn't want to repeat those.  I think you should convert these into one directive that you can add to the `a` elements.

Reply:
@paulbert Please check changes when you have time